sha/ta/include/sha_ta.h

```diff
@@ -0,0 +1,99 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2025, Advanced Micro Devices, Inc. All rights reserved.
+ */
+
+#ifndef __SHA_TA_H__
+#define __SHA_TA_H__
+
+/* UUID of the SHA example trusted application */
+#define TA_SHA_UUID \
+	{ 0x1dc6a16b, 0x2fba, 0x4aa1, \
+		{ 0x95, 0x19, 0xea, 0x8a, 0x6c, 0x8c, 0x16, 0xe5 } }
+
+/*
+ * TA_SHA_CMD_PREPARE - Allocate resources for the SHA operation
+ * param[0] (value) a: TA_ALGO_HMAC_xxx, b: unused
+ * param[1] (value) a: key size in bytes, b: unused
+ * param[2] (value) a: obj_type, b: unused
+ * param[3] unused
+ */
+#define TA_SHA_CMD_PREPARE		0
+
+/* SHA Algo */
+#define TEE_ALG_SHA1			0x50000002
```

Comment:
These definition override the TEE generic definition from **tee_api_defines.h**. Please use IDs dedicated to the TA), e.g.:
```c
enum ta_sha_alog_id  {
	TA_SHA_ALG_INVALID = 0,
	TA_SHA_ALG_SHA1,
	TA_SHA_ALG_SHA224,
	TA_SHA_ALG_SHA256,
	...
```
and add a conversion helper function `TA_SHA_ALG_xxx` => `TAA_ALG_xxx` in the TA implementation.

The inline description at line 16 could be:
```
 * param[0] (value) a: Algorithm ID from enum ta_sha_algo_id, b: unused
```

As-is, many compilers will likely emit build warning message on macro redefinition occurrences.